docs/consumer/VirtualStaking.md

```diff
@@ -0,0 +1,116 @@
+# Virtual Staking
+
+Virtual Staking is a permissioned contract on the Consumer side that can interact with the
+native staking module in special ways. There are usually multiple Virtual Staking contracts
+on one Consumer chain, one for each active Provider, linked 1-to-1 with a Converter.
+
+## Previous Work
+
+There is a lot of overlap with this design and Osmosis' [design of SuperFluid Staking](https://github.com/osmosis-labs/osmosis/tree/main/x/superfluid).
+Before commenting on the technical feasibility or implementation on the SDK side, it would be good to review this in depth.
+This is probably the biggest change made to staking functionality without forking `x/staking` and should be leveraged for the MVP at least.
+
+(Note to author as well as all reviewers)
+
+## Extensibility
+
+Virtual Staking is primarily the interface exposed to the Convert contract, so we can use wildly
+different implementations of it on different chains. Since we want to limit the customizations,
+I will describe the simplest workable customizations for a Cosmos SDK chain.
+
+Some chains may want to provide more native powers, and extend the Virtual Staking contract with a custom
+one to make use of them. Other architectures, like Picasso or cw-sdk, may have a wildly different architecture
+for interacting with the staking module. That would require a Virtual Staking contract that fulfills the
+high-level design points below, but not the implementation.
+
+To keep things clear for now, I will focus on the standard contract and module we aim to ship as part
+of the basic design.
+
+## Components
+
+Virtual Staking will be a mix of a Cosmos SDK module and a CosmWasm contract. 
+We need to expose some special functionality through the SDK Module and `CustomMsg` type.
```

Comment:
Probably worth noting _why_ we need to expose some special functionality in a custom SDK module:
- Easier setup UX (no need to fund a contract from the community pool like in the prototype)
- Make it easier for existing stakers to cross-stake without having to unbond
- Allow for future options such as enabling mesh-governance or providing Comet BFT voting power without governance voting power

Reply:
> Easier setup UX (no need to fund a contract from the community pool like in the prototype)

It's not easier UX. It's usable UX.

This hackathon demo could be totally broken with the unbonding issue.
I bond tokens, then unbond, then bond, then unbond... and soon the contract won't be able to respond due to SDK limits. So we need complex batching mechanisms, etc.

> Make it easier for existing stakers to cross-stake without having to unbond

This is not included. We only have custom module proposed on the consumer side.
We would need customizations on the provider side to allow that. (Which would be a good point to explore for v1/v2)